### PR TITLE
=per #18162 harden PersistentFSMActorSpec, less timing sensitive

### PR DESCRIPTION
**Pull request by ktoso**
While the last assertion `within(1.9 seconds, 2.9 seconds)` "makes sense" it's very timing sensitive I'd say... Let's avoid it here;

Resovles https://github.com/akka/akka/issues/18162

### CODE REVIEW

**Comment by ktoso** [comment]
I'm contemplating how to make the timeout part of the state object... though that's a bit though.
Maybe the test should listen to debug events and notice the timeout value there.

**Comment by patriknw** [comment]
LGTM, I guess

File: akka-persistence/src/main/scala/akka/persistence/fsm/FSM.scala [hunk 1]
```diff
@@ -33,7 +33,12 @@ object FSM {
    * [[akka.actor.FSM.SubscribeTransitionCallBack]] before sending any
    * [[akka.actor.FSM.Transition]] messages.
    */
-  final case class CurrentState[S](fsmRef: ActorRef, state: S)
+  sealed case class CurrentState[S](fsmRef: ActorRef, state: S) {
+    def timeout: Option[FiniteDuration] = None
+    def withTimeout(_timeout: Option[FiniteDuration]) = new CurrentState(fsmRef, state) {
```
Comment (drewhk) [inline akka-persistence/src/main/scala/akka/persistence/fsm/FSM.scala]: why not a new field? Binary compat?

Comment (ktoso) [inline akka-persistence/src/main/scala/akka/persistence/fsm/FSM.scala]: Yeah, it would break compat since it was a public case class

Comment (ktoso) [inline akka-persistence/src/main/scala/akka/persistence/fsm/FSM.scala]: Actually... it's the `persistence.fsm.FSM`, can be breaking then!

Comment (patriknw) [inline akka-persistence/src/main/scala/akka/persistence/fsm/FSM.scala]: +1
we have not even released this in anything else than in 2.4 milestones

**Comment by drewhk** [comment]
LGTM

**Comment by patriknw** [comment]
LGTM

File: akka-docs/rst/scala/persistence.rst [hunk 2]
```diff
@@ -655,16 +655,16 @@ adaptation simply return ``EventSeq.empty``. The adapted events are then deliver
 
 Persistent FSM
 ==============
-``PersistentFSMActor`` handles the incoming messages in an FSM like fashion.
+``PersistentFSM`` handles the incoming messages in an FSM like fashion.
```
Comment (ktoso) [inline akka-docs/rst/scala/persistence.rst]: cc @leonidb this makes the classes more consistent with `FSM` (and how we refer to them anyway "persistent fsm" :smile:)

File: akka-persistence/src/main/java/akka/persistence/fsm/japi/pf/FSMStateFunctionBuilder.java [hunk 3]
```diff
@@ -4,7 +4,8 @@
 
 package akka.persistence.fsm.japi.pf;
 
-import akka.persistence.fsm.FSM;
+import akka.persistence.fsm.PersistentFSM;
+import akka.persistence.fsm.PersistentFSMBase;
```
Comment (ktoso) [inline akka-persistence/src/main/java/akka/persistence/fsm/japi/pf/FSMStateFunctionBuilder.java]: users never see the `*Base`

File: akka-persistence/src/main/scala/akka/persistence/fsm/PersistentFSMBase.scala [hunk 4]
```diff
@@ -7,212 +7,10 @@ import akka.actor._
 import akka.japi.pf.{ UnitPFBuilder, UnitMatch, FSMTransitionHandlerBuilder }
 
 import language.implicitConversions
-import scala.annotation.varargs
-import scala.concurrent.duration.Duration
 import scala.collection.mutable
 import akka.routing.{ Deafen, Listen, Listeners }
 import scala.concurrent.duration.FiniteDuration
 
-object FSM {
```
Comment (ktoso) [inline akka-persistence/src/main/scala/akka/persistence/fsm/PersistentFSMBase.scala]: This is now all in `object PersistentFSM` to make importing simpler to discover

**Comment by ktoso** [comment]
Added second commit which addresses https://github.com/akka/akka/issues/18168 (as discussed on gitter) 488b1f4d6ddf9e9cc9ab769293979250c5c7d936

**Comment by patriknw** [comment]
rename LGTM

**Pull request merged**